modules/fraction/test/test_fraction.cpp

```diff
@@ -0,0 +1,335 @@
+
+    // Assert
+    EXPECT_EQ(f1, f2);
+}
+
+TEST_F(FractionTest, Addition_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 + n;
+    Fraction f3 = n + f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(76, 15));
+    EXPECT_EQ(f3, Fraction(76, 15));
+}
+
+TEST_F(FractionTest, Subtraction_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 - n;
+    Fraction f3 = n - f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(-74, 15));
+    EXPECT_EQ(f3, Fraction(74, 15));
+}
+
+TEST_F(FractionTest, Multiplication_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 * n;
+    Fraction f3 = n * f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(1, 3));
+    EXPECT_EQ(f3, Fraction(1, 3));
+}
+
+TEST_F(FractionTest, Division_Fractions_With_Int_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    int n = 5;
+
+    // Act
+    Fraction f2 = f1 / n;
+    Fraction f3 = n / f1;
+
+    // Assert
+    EXPECT_EQ(f2, Fraction(1, 75));
+    EXPECT_EQ(f3, Fraction(75, 1));
+}
+
+TEST_F(FractionTest, Fraction_To_Double_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+
+    // Act
+    double d = static_cast<double>(1) / 15;
+
+    // Assert
+    EXPECT_NEAR(static_cast<double>(f1), d, FractionTest::epsilon);
+}
+
+TEST_F(FractionTest, Addition_Fractions_With_Double_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    double n1 = 5.0;
+
+    // Act
+    double n2 = static_cast<double>(f1) + n1;
+    double n3 = n1 + static_cast<double>(f1);
+
+    // Assert
+    EXPECT_NEAR(n2, 5.06667, FractionTest::epsilon);
+    EXPECT_NEAR(n3, 5.06667, FractionTest::epsilon);
+}
+
+TEST_F(FractionTest, Subtraction_Fractions_With_Double_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    double n = 5.0;
+
+    // Act
+    double n2 = static_cast<double>(f1) - n;
+    double n3 = n - static_cast<double>(f1);
+
+    // Assert
+    EXPECT_NEAR(n2, -4.93334, FractionTest::epsilon);
+    EXPECT_NEAR(n3, 4.93334, FractionTest::epsilon);
+}
+
+TEST_F(FractionTest, Multiplication_Fractions_With_Double_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    double n = 5.0;
+
+    // Act
+    double n2 = static_cast<double>(f1) * n;
+    double n3 = n * static_cast<double>(f1);
+
+    // Assert
+    EXPECT_NEAR(n2, 0.33334, FractionTest::epsilon);
+    EXPECT_NEAR(n3, 0.33334, FractionTest::epsilon);
+}
+
+TEST_F(FractionTest, Division_Fractions_With_Double_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    double n = 5.0;
+
+    // Act
+    double n2 = static_cast<double>(f1) / n;
+    double n3 = n / static_cast<double>(f1);
+
+    // Assert
+    EXPECT_NEAR(n2, 0.01334, FractionTest::epsilon);
+    EXPECT_NEAR(n3, 75, FractionTest::epsilon);
+}
+
+TEST_F(FractionTest, Operator_Double_Test) {
+    // Arrange
+    Fraction f1(1, 15);
+    double n1 = static_cast<double>(1) / 15;
```

Comment:
вещественные константы по умолчанию double
```suggestion
    double n1 = 1.0 / 15;
```